packages/berlin/src/components/header/Header.tsx

```diff
@@ -1,53 +1,156 @@
+      <section className="mx-auto flex min-h-16 w-[min(90%,1080px)] items-center justify-between">
+        <div className="flex items-center gap-2">
+          <img src={`/logos/lexicon-${theme}.svg`} alt="Lexicon Logo" height={32} width={32} />
+          <h1 className="text-2xl font-semibold leading-6">Lexicon</h1>
+        </div>
+        <NavigationMenu className="font-raleway uppercase">
+          <NavigationMenuList className="gap-3">
+            {user ? (
+              <>
+                <div className="hidden gap-3 md:flex">
+                  <HeaderLinks user={user} />
+                  <UserMenu />
+                </div>
+                <div className="md:hidden">
+                  <Icon onClick={() => setIsMenuOpen(!isMenuOpen)}>
+                    <Menu />
+                  </Icon>
+                </div>
+              </>
+            ) : (
+              <ZupassLoginButton>Login</ZupassLoginButton>
+            )}
+            <NavigationMenuItem>
+              <NavigationMenuLink>
+                <ThemeToggler />
+              </NavigationMenuLink>
+            </NavigationMenuItem>
+          </NavigationMenuList>
+        </NavigationMenu>
+      </section>
+    </header>
+  );
+}
 
-// API
-import { logout } from 'api';
+const HeaderLinks = ({ user }: { user: GetUserResponse }) => {
+  const { data: events } = useQuery({
+    queryKey: ['events'],
+    queryFn: fetchEvents,
+    enabled: !!user,
+  });
 
-// Hooks
-import useUser from '../../hooks/useUser';
+  const { data: registrationsData } = useQuery({
+    queryKey: [user?.id, 'registrations'],
+    queryFn: () => fetchUserRegistrations(user?.id ?? ''),
+    enabled: !!user,
+  });
 
-// Components
-import Button from '../button';
-import NavButton from '../nav-button';
-import ZupassLoginButton from '../zupass-button/ZupassLoginButton';
+  const { data: alerts } = useQuery({
+    queryKey: ['alerts'],
+    queryFn: () => fetchAlerts(),
+    enabled: !!user,
+    refetchInterval: 10000,
+  });
 
-// Styled components
-import {
-  Bar,
-  BurgerMenuContainer,
-  DesktopButtons,
-  HeaderContainer,
-  LogoContainer,
-  LogoSubtitle,
-  LogoTextContainer,
-  LogoTitle,
-  MenuButton,
-  MobileButtons,
-  NavButtons,
-  NavContainer,
-  StyledHeader,
-} from './Header.styled';
+  const links = useMemo(() => {
+    const baseLinks = [
+      {
+        title: 'My Proposals',
+        link: events ? `/events/${events?.[0].id}/register` : '',
+      },
+      {
+        title: 'Agenda',
+        link: events ? `/events/${events?.[0].id}/cycles` : '',
+      },
+    ];
+
+    if (
+      registrationsData?.some((registration) => registration.status === 'APPROVED') &&
+      alerts &&
+      alerts.length > 0
+    ) {
+      const alertsLinks = alerts.map((alert) => ({
+        title: alert.title,
+        link: alert.link || '',
+      }));
+      return [...baseLinks, ...alertsLinks];
+    }
+
+    return baseLinks;
+  }, [events, registrationsData, alerts]);
```

Comment:
i like this, we might have to make my proposals and agenda an alert link at some point as it really only makes sense inside an event. but this is a good vision and refactor